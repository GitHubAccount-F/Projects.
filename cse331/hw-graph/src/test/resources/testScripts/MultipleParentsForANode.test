#Tests to see that a node
#can have multiple parent nodes.
#We also test that this works when a node is
#in between two other nodes.

#Creates empty graphs
CreateGraph graph1
CreateGraph graph2
CreateGraph graph3
CreateGraph graph4

#Adds a node
AddNode graph2 one
AddNode graph2 two
AddNode graph3 one
AddNode graph3 two
AddNode graph3 three
AddNode graph4 one
AddNode graph4 two
AddNode graph4 three
AddNode graph4 four

#Adds edges
AddEdge graph2 one two e1
AddEdge graph3 one two e1
AddEdge graph3 three two e1
AddEdge graph4 one three e1
AddEdge graph4 two three e2
AddEdge graph4 three four e3

#Gives information about child nodes, which can use to then see
#how many parents a node has
ListChildren graph1 one
ListChildren graph2 one
ListChildren graph3 one
ListChildren graph3 three
ListChildren graph4 one
ListChildren graph4 two
ListChildren graph4 three