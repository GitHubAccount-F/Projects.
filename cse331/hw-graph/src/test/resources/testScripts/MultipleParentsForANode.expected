#Tests to see that a node
#can have multiple parent nodes.
#We also test that this works when a node is
#in between two other nodes.

#Creates empty graphs
created graph graph1
created graph graph2
created graph graph3
created graph graph4

#Adds a node
added node one to graph2
added node two to graph2
added node one to graph3
added node two to graph3
added node three to graph3
added node one to graph4
added node two to graph4
added node three to graph4
added node four to graph4

#Adds edges
added edge e1 from one to two in graph2
added edge e1 from one to two in graph3
added edge e1 from three to two in graph3
added edge e1 from one to three in graph4
added edge e2 from two to three in graph4
added edge e3 from three to four in graph4

#Gives information about child nodes, which can use to then see
#how many parents a node has
the children of one in graph1 are:
the children of one in graph2 are: two(e1)
the children of one in graph3 are: two(e1)
the children of three in graph3 are: two(e1)
the children of one in graph4 are: three(e1)
the children of two in graph4 are: three(e2)
the children of three in graph4 are: four(e3)